exchanges/order/order_test.go

```diff
@@ -400,71 +351,42 @@ func TestInferCostsAndTimes(t *testing.T) {
 	detail.Amount = 1
 	detail.RemainingAmount = 1
 	detail.InferCostsAndTimes()
-	if detail.Amount != detail.ExecutedAmount+detail.RemainingAmount {
-		t.Errorf(
-			"Order detail amounts not equals. Expected 0, received %f",
-			detail.Amount-(detail.ExecutedAmount+detail.RemainingAmount),
-		)
-	}
+	assert.Equal(t, detail.ExecutedAmount+detail.RemainingAmount, detail.Amount)
 	detail.RemainingAmount = 0
 
 	detail.Amount = 1
 	detail.ExecutedAmount = 1
 	detail.Price = 2
 	detail.InferCostsAndTimes()
-	if detail.AverageExecutedPrice != 2 {
-		t.Errorf(
-			"Unexpected AverageExecutedPrice. Expected 2, received %f",
-			detail.AverageExecutedPrice,
-		)
-	}
+	assert.Equal(t, 2., detail.AverageExecutedPrice)
 
 	detail = Detail{Amount: 1, ExecutedAmount: 2, Cost: 3, Price: 0}
 	detail.InferCostsAndTimes()
-	if detail.AverageExecutedPrice != 1.5 {
-		t.Errorf(
-			"Unexpected AverageExecutedPrice. Expected 1.5, received %f",
-			detail.AverageExecutedPrice,
-		)
-	}
+	assert.Equal(t, 1.5, detail.AverageExecutedPrice)
 
 	detail = Detail{Amount: 1, ExecutedAmount: 2, AverageExecutedPrice: 3}
 	detail.InferCostsAndTimes()
-	if detail.Cost != 6 {
-		t.Errorf(
-			"Unexpected Cost. Expected 6, received %f",
-			detail.Cost,
-		)
-	}
+	assert.Equal(t, 6., detail.Cost)
 }
 
 func TestFilterOrdersByType(t *testing.T) {
 	t.Parallel()
 
 	orders := []Detail{
-		{
-			Type: ImmediateOrCancel,
-		},
 		{
 			Type: Limit,
 		},
 		{}, // Unpopulated fields are preserved for API differences
 	}
 
 	FilterOrdersByType(&orders, AnyType)
-	if len(orders) != 3 {
-		t.Errorf("Orders failed to be filtered. Expected %v, received %v", 2, len(orders))
-	}
+	assert.Lenf(t, orders, 2, "Orders failed to be filtered. Expected %v, received %v", 2, len(orders))
```

Comment:
```suggestion
	assert.Len(t, orders, 2, "Orders should be filtered correctly")
```

Since this is currently redundant as assert will mention the amount expected vs received, same for the others below
![image](https://github.com/user-attachments/assets/57a93129-7e0c-4d31-bb52-a0c8d3d5c59e)